todoman/ui.py

```diff
@@ -150,12 +153,7 @@ def _save_inner(self):
 
         self.todo.is_completed = self._completed.get_state()
 
-        # If it was already non-zero, keep it that way. Let's not overwrite
-        # values 1 thru 8.
-        if self._urgent.get_state() and not self.todo.priority:
-            self.todo.priority = 9
-        elif not self._urgent.get_state():
-            self.todo.priority = 0
+        self.todo.priority = self.formatter.parse_priority(self._priority)
```

Comment:
This doesn't work, because `self._priority` is a ExtendedEdit, not a text by itself. You need to add a property like this further below:

    @property
    def priority(self):
        return self._priority.edit_text

and call parse_priority with `self.priority` instead of `self._priority`